File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
     int[] arr = new int[nums.length];
     int positive = 0;
     int negative = 1;
     for(int i = 0; i < nums.length; i++) {
         if(nums[i] > 0) {
             arr[positive] = nums[i];
             positive+=2;
         }else{
             arr[negative] = nums[i];
             negative+=2;
         }
     } 
        return arr;
    }
}

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int n=nums.size();
        vector<int >ans(n,0);
        int evenIndex =0 ,oddInd=1;
        for(auto x :nums){
            if(x>0){
                ans[evenIndex]=x;
                evenIndex+=2;
            }else if( x<0){
                ans[oddInd]=x;
                oddInd+=2;
            }
        }
        return ans;
    }
};